backend/app/models/authorization.py

```diff
@@ -20,9 +22,10 @@ class RoleType(str, Enum):
 class AuthorizationBase(BaseModel):
     # TODO: This should be PyObjectId = Field(default_factory=PyObjectId). Need to figure out why can't create instance
     #  in `routers.authorization.get_dataset_role()`.
-    dataset_id: str
-    user_id: EmailStr
+    dataset_id: PyObjectId
+    user_ids: List[EmailStr] = []
```

Comment:
Ah... we are doing a list of user_ids now? this might break the view :-( cuz I was assuming this is just one string

Reply:
If we have decided to make user_ids a list, then I'll need to update some of the logic to filter user access. Will convert this into an issue.